RecoVertex/PrimaryVertexProducer/interface/WeightedMeanFitter.h

```diff
@@ -0,0 +1,468 @@
+        edm::LogWarning("WeightedMeanFitter")
+            << "Vertex fitting failed, either all tracks are outliers or they have a very large error\n";
+        return TransientVertex(GlobalPoint(0, 0, 0), err, iclus, 0, 0);
+      }
+      wx = bse.cxx() <= std::pow(precision, 2) ? 1. / std::pow(precision, 2) : 1. / bse.cxx();
+      wy = bse.cyy() <= std::pow(precision, 2) ? 1. / std::pow(precision, 2) : 1. / bse.cyy();
+
+      x += bsp.x() * wx;
+      y += bsp.y() * wy;
+      s_wx += wx;
+      s2_wx += wx;
+      s_wy += wy;
+      s2_wy += wy;
+
+      x /= s_wx;
+      y /= s_wy;
+      z /= s_wz;
+
+      err_x = (s2_wx / std::pow(s_wx, 2));
+      err_y = (s2_wy / std::pow(s_wy, 2));
+      err_z = (s2_wz / std::pow(s_wz, 2));
+
+      if (std::abs(x - old_x) < (precision) && std::abs(y - old_y) < (precision) && std::abs(z - old_z) < (precision)) {
+        break;
+      }
+    }
+    err(0, 0) = err_x * corr_x_bs * corr_x_bs;
+    err(1, 1) = err_y * corr_x_bs * corr_x_bs;
+    err(2, 2) = err_z * corr_z * corr_z;
+
+    float dist = 0;
+    for (const auto& p : points) {
+      wx = p.second.x();
+      wx = wx <= precision ? precision : wx;
+
+      wz = p.second.z();
+      wz = wz <= precision ? precision : wz;
+
+      dist = std::pow(p.first.x() - x, 2) / (std::pow(wx, 2) + err(0, 0));
+      dist += std::pow(p.first.y() - y, 2) / (std::pow(wx, 2) + err(1, 1));
+      dist += std::pow(p.first.z() - z, 2) / (std::pow(wz, 2) + err(2, 2));
+      chi2 += dist;
+    }
+    TransientVertex v(GlobalPoint(x, y, z), err, iclus, chi2, (int)ndof_x);
+    return v;
+  }
+
+  inline TransientVertex weightedMeanOutlierRejectionVarianceAsError(
+      const std::vector<std::pair<GlobalPoint, GlobalPoint>>& points,
+      std::vector<std::vector<reco::TransientTrack>>::const_iterator iclus) {
+    float x = 0, y = 0, z = 0, s_wx = 0, s_wy = 0, s_wz = 0, s2_wx = 0, s2_wy = 0, s2_wz = 0, wx = 0, wy = 0, wz = 0,
+          chi2 = 0;
+    float ndof_x = 0;
+    AlgebraicSymMatrix33 err;
+    err(2, 2) = startError * startError;  // error is 20 cm, so cov -> is 20 ^ 2
+    err(0, 0) = err(1, 1) = err(2, 2) / 100.;
+
+    for (const auto& p : points) {
+      wx = p.second.x();
+      wx = wx <= precision ? 1. / std::pow(precision, 2) : 1. / std::pow(wx, 2);
+
+      wz = p.second.z();
+      wz = wz <= precision ? 1. / std::pow(precision, 2) : 1. / std::pow(wz, 2);
+
+      x += p.first.x() * wx;
+      y += p.first.y() * wx;
+      z += p.first.z() * wz;
+
+      s_wx += wx;
+      s_wz += wz;
+    }
+
+    if (s_wx == 0. || s_wz == 0.) {
+      edm::LogWarning("WeightedMeanFitter") << "Vertex fitting failed at beginning\n";
+      return TransientVertex(GlobalPoint(0, 0, 0), err, *iclus, 0, 0);
+    }
+
+    x /= s_wx;
+    y /= s_wx;
+    z /= s_wz;
+
+    float old_x, old_y, old_z;
+    float xpull;
+    int niter = 0;
+    float mu = 3.;
+    float err_x, err_y, err_z;
+    err_x = 1. / std::sqrt(s_wx);
```

Comment:
These err_x,y,z are only used squared inside the loop below, and then redefined right after.
They could then more conveninently defined here as
```suggestion
    err_x = 1. / s_wx;
```
etc.